### PR TITLE
tests: added log max test to ctr.bats and command.bats

### PR DESCRIPTION
**Pull request by chuanchang**
$ bats -f "ctr log max with" test/ctr.bats
 ✓ ctr log max with default value
 ✓ ctr log max with minimum value

2 tests, 0 failures

$ bats -f "log max" test/command.bats 
 ✓ invalid log max
 ✓ log max boundary testing

2 tests, 0 failures

Signed-off-by: Alex Jia <chuanchang.jia@gmail.com>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by giuseppe** [comment]
LGTM

File: test/command.bats [hunk 1]
```diff
@@ -37,4 +37,29 @@ load helpers
...
+	run ${CRIO_BINARY} --log-size-max 8191
+	echo $output
+	[ "$status" -ne 0 ]
+	[[ "$output" =~ "log size max should be negative or >= 8192" ]]
+	# log size max is out of the range of 64-bit signed integers
+	run ${CRIO_BINARY} --log-size-max 18446744073709551616
+	echo $output
+	[ "$status" -ne 0 ]
+	[[ "$output" =~ "value out of range" ]]
+}
```
Comment (TomSweeneyRedHat) [inline test/command.bats]: Might be good to add a test for 8192 and -1 too.

Comment (TomSweeneyRedHat) [reply test/command.bats]: oops, nvm, looks like you do so below.

**Comment by saschagrunert** [comment]
LGTM

**Comment by umohnani8** [comment]
@chuanchang the test is failing on travis https://travis-ci.com/cri-o/cri-o/jobs/193527292#L4571

**Comment by chuanchang** [comment]
> @chuanchang the test is failing on travis https://travis-ci.com/cri-o/cri-o/jobs/193527292#L4571

the /etc/crio/crio.conf is not necessary for testing, because test case is starting crio w/ cmdline and command option directly, a fatal error is "config validation: invalid conmon path: stat /usr/local/libexec/crio/conmon: no such file or directory",  I will check it further.

```shell
ok 10 invalid metrics port
ok 11 invalid log max
not ok 12 log max boundary testing
# (in test file ./command.bats, line 65)
#   `[[ "$output" =~ "log size max should be negative or >= 8192" ]]' failed
# time="2019-04-18T13:47:19Z" level=warning msg="default configuration file does not exist: /etc/crio/crio.conf" time="2019-04-18 13:47:19.779032053Z" level=fatal msg="config validation: invalid conmon path: stat /usr/local/libexec/crio/conmon: no such file or directory"
```

**Comment by rhatdan** [comment]
@chuanchang Any movement on this?

**Comment by saschagrunert** [comment]
@chuanchang could you please rebase on top of the latest master branch? We did some test isolation enhancements there with the implementation of the parallel test execution.

**Comment by chuanchang** [comment]
> @chuanchang could you please rebase on top of the latest master branch? We did some test isolation enhancements there with the implementation of the parallel test execution.

Sure, I hope everything is okay this time ;-)

**Comment by chuanchang** [comment]
> @chuanchang Any movement on this?

Hi Daniel, I just came back from vacation and sorry to update this late, I have rebased this patch on top of the latest master branch.

**Comment by chuanchang** [comment]
@saschagrunert I'm not sure how to solve the following failures of Travis CI.

```shell
prog.go:12:2: cannot find package "github.com/golang/mock/mockgen/model" in any of:
	/Users/travis/.gimme/versions/go1.11.9.darwin.amd64/src/github.com/golang/mock/mockgen/model (from $GOROOT)
	/Users/travis/gopath/src/github.com/golang/mock/mockgen/model (from $GOPATH)
2019/05/05 01:48:13 Loading input failed: exit status 1
make: *** [mockgen] Error 1
The command "make testunit" exited with 2.
2.00s$ make
touch "/Users/travis/gopath/.gopathok"
go build -ldflags '-s -w -X main.gitCommit="f2124e5cdac5c47d23c7f8fd631911ef907032d6" -X main.buildInfo=1557020893' -tags "containers_image_ostree_stub  exclude_graphdriver_btrfs btrfs_noversion exclude_graphdriver_devicemapper libdm_no_deferred_remove   " -o bin/crio github.com/cri-o/cri-o/cmd/crio
lib/sandbox/sandbox.go:10:2: build constraints exclude all Go files in /Users/travis/gopath/src/github.com/cri-o/cri-o/vendor/github.com/containernetworking/plugins/pkg/ns
vendor/github.com/containers/buildah/common.go:8:2: C source files not allowed when not using cgo or SWIG: unshare.c
make: *** [bin/crio] Error 1
The command "make" exited with 2.

Package ostree-1 was not found in the pkg-config search path.
Perhaps you should add the directory containing `ostree-1.pc'
to the PKG_CONFIG_PATH environment variable
No package 'ostree-1' found
pkg-config: exit status 1
Makefile:170: recipe for target 'bin/crio.cross.linux.amd64' failed
make[1]: *** [bin/crio.cross.linux.amd64] Error 2
make[1]: Leaving directory '/home/travis/gopath/src/github.com/cri-o/cri-o'
Makefile:167: recipe for target 'local-cross' failed
make: *** [local-cross] Error 2
The command "make local-cross" exited with 2.

ok 116 crictl runtimeversion
Using cri-o binary '/go/src/github.com/cri-o/cri-o/bin/crio'
ok 117 ctr termination reason Completed
Using cri-o binary '/go/src/github.com/cri-o/cri-o/bin/crio'
ok 118 ctr check shared /dev/shm
real	9m47.547s
user	6m34.122s
sys	2m28.912s
Makefile:289: recipe for target 'localintegration' failed
make: *** [localintegration] Error 1
Makefile:183: recipe for target 'integration' failed
make: *** [integration] Error 2
The command "make integration CONTAINER_RUNTIME=$CONTAINER_RUNTIME" exited with 2.
```

File: test/command.bats [hunk 2]
```diff
@@ -49,3 +49,28 @@ load helpers
 	[ "$status" -ne 0 ]
 	[[ "$output" =~ "value out of range" ]]
 }
+
+@test "invalid log max" {
+	run ${CRIO_BINARY} --log-size-max foo
```
Comment (saschagrunert) [inline test/command.bats]: You still have to provide the conmon path to make the test succeed:
```suggestion
	run ${CRIO_BINARY} --conmon "$TESTDIR/conmon" --log-size-max foo
```

File: test/command.bats [hunk 3]
```diff
@@ -49,3 +49,28 @@ load helpers
...
+	echo $output
+	[ "$status" -ne 0 ]
+	[[ "$output" =~ "invalid value \"foo\" for flag" ]]
+}
+
+@test "log max boundary testing" {
+	# log size max is special zero value
+	run ${CRIO_BINARY} --log-size-max 0
```
Comment (saschagrunert) [inline test/command.bats]: The same for this:
```suggestion
	mkdir -p "$TESTDIR/cni/net.d"
	run ${CRIO_BINARY} --conmon "$TESTDIR/conmon" --cni-config-dir "$TESTDIR/cni/net.d" --log-size-max foo
```

File: test/command.bats [hunk 4]
```diff
@@ -49,3 +49,28 @@ load helpers
...
+	echo $output
+	[ "$status" -ne 0 ]
+	[[ "$output" =~ "log size max should be negative or >= 8192" ]]
+	# log size max is less than 8192 and more than 0
+	run ${CRIO_BINARY} --log-size-max 8191
```
Comment (saschagrunert) [inline test/command.bats]: And here too:
```suggestion
	run ${CRIO_BINARY} --conmon "$TESTDIR/conmon" --cni-config-dir "$TESTDIR/cni/net.d" --log-size-max 8191
```

File: test/command.bats (hunk 2 above)
Comment (chuanchang) [reply test/command.bats]: thanks, done

**Comment by chuanchang** [comment]
another failures:

```shell
TASK [run integration tests] ***************************************************
task path: /go/src/github.com/cri-o/cri-o/contrib/test/integration/test.yml:21
fatal: [localhost]: FAILED! => {
    "ansible_job_id": "426884085280.15439", 
    "changed": true, 
    "cmd": "make localintegration >& /tmp/artifacts/testout.txt", 
    "delta": "0:05:14.365812", 
    "end": "2019-05-16 02:15:50.016554", 
    "finished": 1, 
    "rc": 2, 
    "start": "2019-05-16 02:10:35.650742"
}

MSG:
---
non-zero return code
---

--------------------------------------------
TASK [All persistent swap devices from fstab are known] ************************
task path: /go/src/github.com/cri-o/cri-o/contrib/test/integration/disable_swap.yml:12
[DEPRECATION WARNING]: Using tests as filters is deprecated. Instead of using 
`result|failed` instead use `result is failed`. This feature will be removed in
 version 2.9. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
ok: [localhost] => {
    "changed": false, 
    "cmd": [
        "findmnt", 
        "--fstab", 
        "--type", 
        "swap", 
        "--output", 
        "SOURCE", 
        "--noheadings"
    ], 
    "delta": "0:00:00.003343", 
    "end": "2019-05-16 02:09:20.993336", 
    "failed_when_result": false, 
    "rc": 1, 
    "start": "2019-05-16 02:09:20.989993"
}

MSG:
---
non-zero return code
---

----------------------------------------------
TASK [Disable selinux during integration tests] ********************************
task path: /go/src/github.com/cri-o/cri-o/contrib/test/integration/test.yml:17
skipping: [localhost] => {
    "changed": false, 
    "skip_reason": "Conditional result was False"
}

TASK [run integration tests] ***************************************************
task path: /go/src/github.com/cri-o/cri-o/contrib/test/integration/test.yml:21
fatal: [localhost]: FAILED! => {
    "ansible_job_id": "426884085280.15439", 
    "changed": true, 
    "cmd": "make localintegration >& /tmp/artifacts/testout.txt", 
    "delta": "0:05:14.365812", 
    "end": "2019-05-16 02:15:50.016554", 
    "finished": 1, 
    "rc": 2, 
    "start": "2019-05-16 02:10:35.650742"
}

MSG:
---
non-zero return code
---

```

**Comment by haircommander** [comment]
```
not ok 13 log max boundary testing
# (in test file ./command.bats, line 71)
#   `[[ "$output" =~ "log size max should be negative or >= 8192" ]]' failed
# time="2019-05-16 02:10:16.929710195Z" level=fatal msg="library config validation: runtime config: invalid conmon path: stat \"$TESTDIR/conmon\": no such file or directory"
```
@chuanchang I am not sure, but I think by wrapping opt in single quotes you're telling bash to interpret conmon's path as exactly $TESTDIR/conmon instead of with $TESTDIR expanded. 

If you need a bash helper (to create the opts for the tests that just need a simple configuration, you can add it in test/helpers.bash

I'd suggest 
```
function get_minimal_crio_args() {
        mkdir -p "$TESTDIR/cni/net.d"
	echo --conmon "$TESTDIR/conmon" --cni-config-dir "$TESTDIR/cni/net.d
}
```
then you can set opts=$(get_minmal_crio_args) where you need it

**Comment by chuanchang** [comment]
> ```
> not ok 13 log max boundary testing
> # (in test file ./command.bats, line 71)
> #   `[[ "$output" =~ "log size max should be negative or >= 8192" ]]' failed
> # time="2019-05-16 02:10:16.929710195Z" level=fatal msg="library config validation: runtime config: invalid conmon path: stat \"$TESTDIR/conmon\": no such file or directory"
> ```
> 
> @chuanchang I am not sure, but I think by wrapping opt in single quotes you're telling bash to interpret conmon's path as exactly $TESTDIR/conmon instead of with $TESTDIR expanded.

It works well on my Fedora 30, so I think it's not a single quotes issue.
> 
> If you need a bash helper (to create the opts for the tests that just need a simple configuration), you can add it in test/helpers.bash
> 
> I'd suggest
> 
> ```
> function get_minimal_crio_args() {
>         mkdir -p "$TESTDIR/cni/net.d"
> 	echo --conmon "$TESTDIR/conmon" --cni-config-dir "$TESTDIR/cni/net.d
> }
> ```
> 
> then you can set opts=$(get_minmal_crio_args) where you need it

If it's not single quotes issue, perhaps it's not necessary for above changes, but I can update according to your advise, thanks a lot!

**Comment by haircommander** [comment]
/retest

On Wednesday, June 12, 2019, OpenShift CI Robot <notifications@github.com>
wrote:
> @chuanchang: The following test failed, say /retest to rerun them all:
>
> Test name Commit Details Rerun command
> ci/openshift-jenkins/e2e_rhel f327fa1 link /test e2e_rhel
>
> Full PR test history. Your PR dashboard. Please help us cut down on
flakes by linking to an open issue when you hit one in your PR.
>
> Instructions for interacting with me using PR comments are available
here. If you have questions or suggestions related to my behavior, please
file an issue against the kubernetes/test-infra repository. I understand
the commands that are listed here.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub, or mute the thread.<
https://ci3.googleusercontent.com/proxy/WvIqzgjTF-Teh7o8-0tCD_f5EK6Vqsfc3lgDCwcqMXIGdJc1v9ZG-hAtH4oQjDKyCMjNrpKp0sWWT-nD75ZtgzbQh60PCvK-0UImM3P9MY7xjM4GvvO5OtN8e4dw-OQDy8D7eIlsBsINURW_0V24qd-UOv5OAiVfDloNHH5LmKDU2wuOwHRwu3BCzkhtOYxMoNB9oSUZBqtgRU9lAeF8bJCcduInQ1iYWc0VzWSBmA=s0-d-e1-ft#https://github.com/notifications/beacon/ACFJQJYJQMZJXSHGUXH5B2TP2FXDLA5CNFSM4HF6EUBKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXR5TXY.gif>

**Pull request merged**